### PR TITLE
feat(clinical-details) loader shows when fetching and saving form data

### PR DESCRIPTION
**Pull request by NaamanTsur**
This PR continues the axios change of setting is loading manually instead of on every axios request.

### CODE REVIEW

File: client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/useClinicalDetails.ts [hunk 1]
```diff
@@ -1,16 +1,20 @@
 import React from 'react';
 import { useSelector } from 'react-redux';
 
-import axios from 'Utils/axios';
+import axios from 'axios';
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/useClinicalDetails.ts:4]: need to be in the first section

File: client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/useClinicalDetails.ts [hunk 2]
```diff
@@ -184,15 +192,39 @@ const useClinicalDetails = (parameters: useClinicalDetailsIncome): useClinicalDe
...
+                    setIsLoading(false);
                 }
             }
         );
     };
 
-    const saveClinicalDetails = (clinicalDetails: ClinicalDetailsData, epidemiologyNumber: number, investigatedPatientId: number): Promise<void> => {
-        return axios.post('/clinicalDetails/saveClinicalDetails', ({ clinicalDetails: { ...clinicalDetails, epidemiologyNumber, investigatedPatientId } }));
+    const saveClinicalDetails = (clinicalDetails: ClinicalDetailsData, validationDate: Date, id: number): void => {
+
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/useClinicalDetails.ts:205]: delete row

**Review by ItayBenMoshe — CHANGES_REQUESTED** [review]

File: client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/useClinicalDetails.ts (hunk 1 above)
Comment (NaamanTsur) [reply client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/useClinicalDetails.ts:3]: fixed

File: client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/useClinicalDetails.ts (hunk 2 above)
Comment (NaamanTsur) [reply client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/useClinicalDetails.ts]: fixed

**Review by ItayBenMoshe — APPROVED** [review]

**Review by nevobonneh — APPROVED** [review]

**Pull request merged**